### ISSUE TITLE
Bug: no-invalid-label-refs reports escaped label references

### ISSUE THREAD

**Issue by KumJungMin** [eslint/markdown#717]
### Environment

ESLint version: 10.8.0
@eslint/markdown version: 8.0.3
Node version: 22.18.0
npm version: 10.9.3

### Which language are you using?

commonmark

### What did you do?

<details>
<summary>Configuration</summary>

```js
import { defineConfig } from "eslint/config";
import markdown from "@eslint/markdown";

export default defineConfig([
    {
        files: ["**/*.md"],
        plugins: {
            markdown,
        },
        language: "markdown/commonmark",
        rules: {
            "markdown/no-invalid-label-refs": "error",
        },
    },
]);
```

</details>

```md
<!-- eslint markdown/no-invalid-label-refs: "error" -->

<!-- OK -->
\[foo][ ]
\\\[foo][ ]

<!-- NOT OK -->
[foo][ ]
\\[foo][ ]
\\\\[foo][ ]
```

### What did you expect to happen?

he rule should not report the cases with an odd number of backslashes before the opening bracket:
```md
\[foo][ ]
\\\[foo][ ]
```
In these cases, the opening bracket is escaped, so the text does not form a label reference.

The rule should report only the cases where the opening bracket is not escaped:
```md
[foo][ ]
\\[foo][ ]
\\\\[foo][ ]
```
An even number of backslashes does not escape the opening bracket, 
so these cases still form invalid label references.

### What actually happened?

no-invalid-label-refs reports the cases where the opening bracket is escaped:
```
   5:7   error  Label reference 'foo' is invalid due to white space between [ and ]  markdown/no-invalid-la...
```
The rule appears to detect the `][ ]` portion and search backward for an opening bracket without checking whether that bracket is escaped. As a result, escaped text is incorrectly treated as an invalid label reference.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/vitejs-vite-ddfg51us?file=eslint.config.js,test.md

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

> Disclosure: I'm a participant of [open source contribution program OSSCA](https://github.com/eslint-ossca)

A related case was addressed in #490, which updated several rules to distinguish escaped syntax based on whether it is preceded by an odd or even number of consecutive backslashes.

That change covered no-missing-label-refs, no-reversed-media-syntax, and no-missing-atx-heading-space. However, no-invalid-label-refs appears to perform a separate label-reference check that does not account for whether the opening bracket is escaped.

**Comment by lumirlumir** [comment]
> Disclosure: I'm a participant of [open source contribution program OSSCA](https://github.com/eslint-ossca): confirmed.

I can reproduce it and confirm that it’s a bug.

Actually, from a historical perspective, I noticed this bug while working on https://github.com/eslint/markdown/pull/490 and several other similar fixes. However, this one involved some additional edge cases related to handling consecutive backslashes, which made it quite tricky at the time. I didn’t have enough time to investigate it further, so I left it as is.

Another edge case to consider for this bug would be:

```md
[foo\][]

[foo]: hi
```

Looking at the Code Explorer, cases with an odd number of backslashes should not be reported. Only cases with an even number of backslashes should be reported.

- [Recognized as `LinkReference` and `Definition`](https://explorer.eslint.org/#eslint-explorer=v2.eyJzdGF0ZSI6eyJ0b29sIjoiYXN0IiwiY29kZSI6eyJqYXZhc2NyaXB0IjoiY29uc3QgaGkgPSA0MjtcblxuZGVidWdnZXI7XG5cbmRlYnVnZ2VyO1xuXG5FU0xpbnQiLCJqc29uIjoiIiwibWFya2Rvd24iOiJbZm9vXVtdXG5cbltmb29dOiBoaSIsImNzcyI6IkBrZXlmcmFtZXMgdGVzdCB7XG5cdDAlIHtcblx0XHRvcGFjaXR5OiAwO1xuXHR9XG5cbiAgICBmcm9tIHtcblx0XHRvcGFjaXR5OiAwO1xuICAgIH1cblxuXHQxMDAlIHtcblx0XHRvcGFjaXR5OiAxO1xuXHR9XG59IiwiaHRtbCI6IjwhRE9DVFlQRSBodG1sPlxuPCEtLVxuVHlwZSBvciBwYXN0ZSBzb21lIEhUTUwgaGVyZSB0byBsZWFybiBtb3JlIGFib3V0XG50aGUgc3RhdGljIGFuYWx5c2lzIHRoYXQgRVNMaW50IGNhbiBkbyBmb3IgeW91LlxuXG5UaGUgdGFicyBhcmU6XG5cbi0gQVNUIC0gVGhlIEFic3RyYWN0IFN5bnRheCBUcmVlIG9mIHRoZSBjb2RlLCB3aGljaCBjYW5cbmJlIHVzZWZ1bCB0byB1bmRlcnN0YW5kIHRoZSBzdHJ1Y3R1cmUgb2YgdGhlIGNvZGUuIFlvdVxuY2FuIHZpZXcgdGhpcyBzdHJ1Y3R1cmUgYXMgSlNPTiBvciBpbiBhIHRyZWUgZm9ybWF0LlxuLS0-XG5cbjxodG1sIGxhbmc9XCJlblwiPlxuICAgIDxoZWFkPlxuICAgICAgICA8bWV0YSBjaGFyc2V0PVwiVVRGLThcIj5cbiAgICAgICAgPHRpdGxlPkhUTUw8L3RpdGxlPlxuICAgIDwvaGVhZD5cbiAgICA8Ym9keT5cbiAgICAgICAgPHA-VGV4dDwvcD5cbiAgICA8L2JvZHk-XG48L2h0bWw-In0sImxhbmd1YWdlIjoibWFya2Rvd24iLCJqc09wdGlvbnMiOnsicGFyc2VyIjoiZXNwcmVlIiwic291cmNlVHlwZSI6Im1vZHVsZSIsImlzVmVyc2lvbiI6ImxhdGVzdCIsImlzSlNYIjp0cnVlfSwianNvbk9wdGlvbnMiOnsianNvbk1vZGUiOiJqc29uNSIsImFsbG93VHJhaWxpbmdDb21tYXMiOmZhbHNlfSwiY3NzT3B0aW9ucyI6eyJjc3NNb2RlIjoiY3NzIiwidG9sZXJhbnQiOmZhbHNlfSwibWFya2Rvd25PcHRpb25zIjp7Im1hcmtkb3duTW9kZSI6ImdmbSIsIm1hcmtkb3duRnJvbnRtYXR0ZXIiOiJvZmYiLCJtYXJrZG93bk1hdGgiOmZhbHNlfSwiaHRtbE9wdGlvbnMiOnsidGVtcGxhdGVFbmdpbmVTeW50YXgiOiJub25lIiwiZnJvbnRtYXR0ZXIiOiJvZmYifSwid3JhcCI6dHJ1ZSwidmlld01vZGVzIjp7ImFzdFZpZXciOiJ0cmVlIiwic2NvcGVWaWV3IjoiZmxhdCIsInBhdGhWaWV3IjoiZ3JhcGgifSwicGF0aEluZGV4Ijp7ImluZGV4IjowLCJpbmRleGVzIjoxfSwiZXNxdWVyeVNlbGVjdG9yIjp7InNlbGVjdG9yIjoiIn19LCJ2ZXJzaW9uIjowfQ):

```md
[foo][]

[foo]: hi
```

- [Recognized as `Text` and `Definition`](https://explorer.eslint.org/#eslint-explorer=v2.eyJzdGF0ZSI6eyJ0b29sIjoiYXN0IiwiY29kZSI6eyJqYXZhc2NyaXB0IjoiY29uc3QgaGkgPSA0MjtcblxuZGVidWdnZXI7XG5cbmRlYnVnZ2VyO1xuXG5FU0xpbnQiLCJqc29uIjoiIiwibWFya2Rvd24iOiJcXFtmb29dW11cblxuW2Zvb106IGhpIiwiY3NzIjoiQGtleWZyYW1lcyB0ZXN0IHtcblx0MCUge1xuXHRcdG9wYWNpdHk6IDA7XG5cdH1cblxuICAgIGZyb20ge1xuXHRcdG9wYWNpdHk6IDA7XG4gICAgfVxuXG5cdDEwMCUge1xuXHRcdG9wYWNpdHk6IDE7XG5cdH1cbn0iLCJodG1sIjoiPCFET0NUWVBFIGh0bWw-XG48IS0tXG5UeXBlIG9yIHBhc3RlIHNvbWUgSFRNTCBoZXJlIHRvIGxlYXJuIG1vcmUgYWJvdXRcbnRoZSBzdGF0aWMgYW5hbHlzaXMgdGhhdCBFU0xpbnQgY2FuIGRvIGZvciB5b3UuXG5cblRoZSB0YWJzIGFyZTpcblxuLSBBU1QgLSBUaGUgQWJzdHJhY3QgU3ludGF4IFRyZWUgb2YgdGhlIGNvZGUsIHdoaWNoIGNhblxuYmUgdXNlZnVsIHRvIHVuZGVyc3RhbmQgdGhlIHN0cnVjdHVyZSBvZiB0aGUgY29kZS4gWW91XG5jYW4gdmlldyB0aGlzIHN0cnVjdHVyZSBhcyBKU09OIG9yIGluIGEgdHJlZSBmb3JtYXQuXG4tLT5cblxuPGh0bWwgbGFuZz1cImVuXCI-XG4gICAgPGhlYWQ-XG4gICAgICAgIDxtZXRhIGNoYXJzZXQ9XCJVVEYtOFwiPlxuICAgICAgICA8dGl0bGU-SFRNTDwvdGl0bGU-XG4gICAgPC9oZWFkPlxuICAgIDxib2R5PlxuICAgICAgICA8cD5UZXh0PC9wPlxuICAgIDwvYm9keT5cbjwvaHRtbD4ifSwibGFuZ3VhZ2UiOiJtYXJrZG93biIsImpzT3B0aW9ucyI6eyJwYXJzZXIiOiJlc3ByZWUiLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiaXNWZXJzaW9uIjoibGF0ZXN0IiwiaXNKU1giOnRydWV9LCJqc29uT3B0aW9ucyI6eyJqc29uTW9kZSI6Impzb241IiwiYWxsb3dUcmFpbGluZ0NvbW1hcyI6ZmFsc2V9LCJjc3NPcHRpb25zIjp7ImNzc01vZGUiOiJjc3MiLCJ0b2xlcmFudCI6ZmFsc2V9LCJtYXJrZG93bk9wdGlvbnMiOnsibWFya2Rvd25Nb2RlIjoiZ2ZtIiwibWFya2Rvd25Gcm9udG1hdHRlciI6Im9mZiIsIm1hcmtkb3duTWF0aCI6ZmFsc2V9LCJodG1sT3B0aW9ucyI6eyJ0ZW1wbGF0ZUVuZ2luZVN5bnRheCI6Im5vbmUiLCJmcm9udG1hdHRlciI6Im9mZiJ9LCJ3cmFwIjp0cnVlLCJ2aWV3TW9kZXMiOnsiYXN0VmlldyI6InRyZWUiLCJzY29wZVZpZXciOiJmbGF0IiwicGF0aFZpZXciOiJncmFwaCJ9LCJwYXRoSW5kZXgiOnsiaW5kZXgiOjAsImluZGV4ZXMiOjF9LCJlc3F1ZXJ5U2VsZWN0b3IiOnsic2VsZWN0b3IiOiIifX0sInZlcnNpb24iOjB9):

```md
\[foo][]

[foo]: hi
```

- [Recognized as `LinkReference` and `Definition`](https://explorer.eslint.org/#eslint-explorer=v2.eyJzdGF0ZSI6eyJ0b29sIjoiYXN0IiwiY29kZSI6eyJqYXZhc2NyaXB0IjoiY29uc3QgaGkgPSA0MjtcblxuZGVidWdnZXI7XG5cbmRlYnVnZ2VyO1xuXG5FU0xpbnQiLCJqc29uIjoiIiwibWFya2Rvd24iOiJcXFxcW2Zvb11bXVxuXG5bZm9vXTogaGkiLCJjc3MiOiJAa2V5ZnJhbWVzIHRlc3Qge1xuXHQwJSB7XG5cdFx0b3BhY2l0eTogMDtcblx0fVxuXG4gICAgZnJvbSB7XG5cdFx0b3BhY2l0eTogMDtcbiAgICB9XG5cblx0MTAwJSB7XG5cdFx0b3BhY2l0eTogMTtcblx0fVxufSIsImh0bWwiOiI8IURPQ1RZUEUgaHRtbD5cbjwhLS1cblR5cGUgb3IgcGFzdGUgc29tZSBIVE1MIGhlcmUgdG8gbGVhcm4gbW9yZSBhYm91dFxudGhlIHN0YXRpYyBhbmFseXNpcyB0aGF0IEVTTGludCBjYW4gZG8gZm9yIHlvdS5cblxuVGhlIHRhYnMgYXJlOlxuXG4tIEFTVCAtIFRoZSBBYnN0cmFjdCBTeW50YXggVHJlZSBvZiB0aGUgY29kZSwgd2hpY2ggY2FuXG5iZSB1c2VmdWwgdG8gdW5kZXJzdGFuZCB0aGUgc3RydWN0dXJlIG9mIHRoZSBjb2RlLiBZb3VcbmNhbiB2aWV3IHRoaXMgc3RydWN0dXJlIGFzIEpTT04gb3IgaW4gYSB0cmVlIGZvcm1hdC5cbi0tPlxuXG48aHRtbCBsYW5nPVwiZW5cIj5cbiAgICA8aGVhZD5cbiAgICAgICAgPG1ldGEgY2hhcnNldD1cIlVURi04XCI-XG4gICAgICAgIDx0aXRsZT5IVE1MPC90aXRsZT5cbiAgICA8L2hlYWQ-XG4gICAgPGJvZHk-XG4gICAgICAgIDxwPlRleHQ8L3A-XG4gICAgPC9ib2R5PlxuPC9odG1sPiJ9LCJsYW5ndWFnZSI6Im1hcmtkb3duIiwianNPcHRpb25zIjp7InBhcnNlciI6ImVzcHJlZSIsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJpc1ZlcnNpb24iOiJsYXRlc3QiLCJpc0pTWCI6dHJ1ZX0sImpzb25PcHRpb25zIjp7Impzb25Nb2RlIjoianNvbjUiLCJhbGxvd1RyYWlsaW5nQ29tbWFzIjpmYWxzZX0sImNzc09wdGlvbnMiOnsiY3NzTW9kZSI6ImNzcyIsInRvbGVyYW50IjpmYWxzZX0sIm1hcmtkb3duT3B0aW9ucyI6eyJtYXJrZG93bk1vZGUiOiJnZm0iLCJtYXJrZG93bkZyb250bWF0dGVyIjoib2ZmIiwibWFya2Rvd25NYXRoIjpmYWxzZX0sImh0bWxPcHRpb25zIjp7InRlbXBsYXRlRW5naW5lU3ludGF4Ijoibm9uZSIsImZyb250bWF0dGVyIjoib2ZmIn0sIndyYXAiOnRydWUsInZpZXdNb2RlcyI6eyJhc3RWaWV3IjoidHJlZSIsInNjb3BlVmlldyI6ImZsYXQiLCJwYXRoVmlldyI6ImdyYXBoIn0sInBhdGhJbmRleCI6eyJpbmRleCI6MCwiaW5kZXhlcyI6MX0sImVzcXVlcnlTZWxlY3RvciI6eyJzZWxlY3RvciI6IiJ9fSwidmVyc2lvbiI6MH0)

```md
\\[foo][]

[foo]: hi
```

- [Recognized as `Text` and `Definition`](https://explorer.eslint.org/#eslint-explorer=v2.eyJzdGF0ZSI6eyJ0b29sIjoiYXN0IiwiY29kZSI6eyJqYXZhc2NyaXB0IjoiY29uc3QgaGkgPSA0MjtcblxuZGVidWdnZXI7XG5cbmRlYnVnZ2VyO1xuXG5FU0xpbnQiLCJqc29uIjoiIiwibWFya2Rvd24iOiJcXFxcXFxbZm9vXVtdXG5cbltmb29dOiBoaSIsImNzcyI6IkBrZXlmcmFtZXMgdGVzdCB7XG5cdDAlIHtcblx0XHRvcGFjaXR5OiAwO1xuXHR9XG5cbiAgICBmcm9tIHtcblx0XHRvcGFjaXR5OiAwO1xuICAgIH1cblxuXHQxMDAlIHtcblx0XHRvcGFjaXR5OiAxO1xuXHR9XG59IiwiaHRtbCI6IjwhRE9DVFlQRSBodG1sPlxuPCEtLVxuVHlwZSBvciBwYXN0ZSBzb21lIEhUTUwgaGVyZSB0byBsZWFybiBtb3JlIGFib3V0XG50aGUgc3RhdGljIGFuYWx5c2lzIHRoYXQgRVNMaW50IGNhbiBkbyBmb3IgeW91LlxuXG5UaGUgdGFicyBhcmU6XG5cbi0gQVNUIC0gVGhlIEFic3RyYWN0IFN5bnRheCBUcmVlIG9mIHRoZSBjb2RlLCB3aGljaCBjYW5cbmJlIHVzZWZ1bCB0byB1bmRlcnN0YW5kIHRoZSBzdHJ1Y3R1cmUgb2YgdGhlIGNvZGUuIFlvdVxuY2FuIHZpZXcgdGhpcyBzdHJ1Y3R1cmUgYXMgSlNPTiBvciBpbiBhIHRyZWUgZm9ybWF0LlxuLS0-XG5cbjxodG1sIGxhbmc9XCJlblwiPlxuICAgIDxoZWFkPlxuICAgICAgICA8bWV0YSBjaGFyc2V0PVwiVVRGLThcIj5cbiAgICAgICAgPHRpdGxlPkhUTUw8L3RpdGxlPlxuICAgIDwvaGVhZD5cbiAgICA8Ym9keT5cbiAgICAgICAgPHA-VGV4dDwvcD5cbiAgICA8L2JvZHk-XG48L2h0bWw-In0sImxhbmd1YWdlIjoibWFya2Rvd24iLCJqc09wdGlvbnMiOnsicGFyc2VyIjoiZXNwcmVlIiwic291cmNlVHlwZSI6Im1vZHVsZSIsImlzVmVyc2lvbiI6ImxhdGVzdCIsImlzSlNYIjp0cnVlfSwianNvbk9wdGlvbnMiOnsianNvbk1vZGUiOiJqc29uNSIsImFsbG93VHJhaWxpbmdDb21tYXMiOmZhbHNlfSwiY3NzT3B0aW9ucyI6eyJjc3NNb2RlIjoiY3NzIiwidG9sZXJhbnQiOmZhbHNlfSwibWFya2Rvd25PcHRpb25zIjp7Im1hcmtkb3duTW9kZSI6ImdmbSIsIm1hcmtkb3duRnJvbnRtYXR0ZXIiOiJvZmYiLCJtYXJrZG93bk1hdGgiOmZhbHNlfSwiaHRtbE9wdGlvbnMiOnsidGVtcGxhdGVFbmdpbmVTeW50YXgiOiJub25lIiwiZnJvbnRtYXR0ZXIiOiJvZmYifSwid3JhcCI6dHJ1ZSwidmlld01vZGVzIjp7ImFzdFZpZXciOiJ0cmVlIiwic2NvcGVWaWV3IjoiZmxhdCIsInBhdGhWaWV3IjoiZ3JhcGgifSwicGF0aEluZGV4Ijp7ImluZGV4IjowLCJpbmRleGVzIjoxfSwiZXNxdWVyeVNlbGVjdG9yIjp7InNlbGVjdG9yIjoiIn19LCJ2ZXJzaW9uIjowfQ)

```md
\\\[foo][]

[foo]: hi
```

---

Using the same regular expression pattern as the one below might be helpful in this case:

https://github.com/eslint/markdown/blob/ac31775656f65f78bc02c0a377c0a85bd131f9f8/src/rules/no-reversed-media-syntax.js#L23-L25

https://github.com/eslint/markdown/blob/ac31775656f65f78bc02c0a377c0a85bd131f9f8/src/rules/no-missing-label-refs.js#L46-L47
